alchemist/alchemist-fxui/src/main/kotlin/it/unibo/alchemist/boundary/RectangleExtensions.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2010-2020, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.boundary
+
+import it.unibo.alchemist.boundary.wormhole.interfaces.BidimensionalWormhole
+import it.unibo.alchemist.model.interfaces.Node
+import it.unibo.alchemist.model.interfaces.Position2D
+import javafx.scene.shape.Rectangle
+import java.awt.Point
+
+/**
+ * Returns the nodes intersecting with the caller rectangle.
+ */
+fun <T, P : Position2D<P>> Rectangle.intersectingNodes(
+    nodes: Map<Node<T>, P>,
+    wormhole: BidimensionalWormhole<P>
+): Map<Node<T>, P> = let { area -> nodes.filterValues { wormhole.getViewPoint(it) in area } }
```

Comment:
This looks like a way to convert `this` to `area`...

isn't this the same as `nodes.filterValues { wormhole.getViewPoint(it) in this }`?

Reply:
Yep, not sure why I used let at all